src/KdbOptions.h

```diff
@@ -81,6 +156,54 @@ class KdbOptions
     }
   }
 
+  void PopulateNullMappingOptions( long long index, K dict )
+  {
+    using NM = Options::NullMapping::Type;
+
+    K keys = kK( kK( dict )[index] )[0];
+    K values = kK( kK( dict )[index] )[1];
+    for( auto i = 0ll; i < values->n; ++i ){
+      const std::string key = ToUpper( kS( keys )[i] );
+      if( supported_null_mapping_options.find( key ) == supported_null_mapping_options.end() ){
+        throw InvalidOption(("Unsupported NULL_MAPPING option '" + key + "'").c_str());
+      }
+      K value = kK( values )[i];
+      if( ETraits<NM>::name( NM::INT_16 ) == key && -KH == value->t ){
+        null_mapping_options.int16_null = value->h;
+        null_mapping_options.have_int16 = true;
+      }
+      else if( ETraits<NM>::name( NM::INT_32 ) == key && -KI == value->t ){
+        null_mapping_options.int32_null = value->i;
+        null_mapping_options.have_int32 = true;
+      }
+      else if( ETraits<NM>::name( NM::STRING ) == key && KC == value->t ){
+        null_mapping_options.string_null.assign( (char*)kC( value ), value->n );
+        null_mapping_options.have_string = true;
+      }
+      else if( ETraits<NM>::name( NM::LARGE_STRING ) == key && KC == value->t ){
+        null_mapping_options.large_string_null.assign( (char*)kC( value ), value->n );
+        null_mapping_options.have_large_string = true;
+      }
+      else{

```

Comment:
Include:
```
      case 101:
        // Ignore ::
        break;
 ```
 We may need a generic  null (::) in the values list to ensure it's a mixed list.
 Otherwise the NULL_MAPPING dict could be (enlist `int16)!enlist 0Nh in which case the values would be a KH